### DISCUSSION TITLE
Enable Metal Backend for UPBGE

### DISCUSSION THREAD

**Discussion by hezishah** [UPBGE/upbge#1791] [Ideas/Features for UPBGE 0.3+]
Hi. Currently when enabling Metal Backend on macOS, starting the game only draws one triangle.
Are there plans for enabling that?

**Comment by lordloki** [comment]
Yes, i'm investigating how to fix it

**Comment by hezishah** [comment]
Any updates?

**Comment by MGameBox** [comment]
您的邮件，已经收到，如未能及时阅读还请见谅

**Comment by hezishah** [comment]
Hi.
In latest upbge HEAD, I had to do a compile fix in 
[upbge-compile-fix.patch](https://github.com/UPBGE/upbge/files/15159104/upbge-compile-fix.patch)
 and a runtime fix to prevent runtime crash.
 
[upbge-runtime-fix.patch](https://github.com/UPBGE/upbge/files/15159114/upbge-runtime-fix.patch)

After those fixes, the Game Engine would run, however only ViewPort Rendering works.

Will we have UPBGE standalone render work in the coming future for M1/2/3 Macs?

**Comment by lordloki** [comment]
Thanks for the fixes. In august I will have some free time to try fixing MacOS port.

**Comment by lordloki** [comment]
Last version has enable Metal backend for MacOS. Closing the feature request as is implemented
